syncutil/broadcast.go

```diff
@@ -35,6 +37,8 @@ type broadcast struct {
 	mutex   sync.Mutex
 }
 
+const broadcastChannelSize = 10000
```

Comment:
Can we make this `DefaultChannelSize`, and then give the user the ability to change the size if they want.